src/backend/src/modules/puterai/AIChatService.js

```diff
@@ -419,16 +419,12 @@ class AIChatService extends BaseService {
                     FunctionCalling.normalize_tools_object(parameters.tools);
                 }
 
-                if ( intended_service === this.service_name ) {
-                    throw new Error('Calling ai-chat directly is not yet supported');
-                }
-
                 const svc_driver = this.services.get('driver');
                 let ret, error;
                 let service_used = intended_service;
-                let model_used = this.get_model_from_request(parameters, {
-                    intended_service
-                });
+                let model_used;
+                ({ model: model_used } = parseModelString(parameters.model));
```

Comment:
Where are taking into account `supplier`? How do handle openrouter for example? Also, shouldn't `vendor` be used to determine the appropriate service for other cases?